client/state/plugins/installed/selectors-ts.ts

```diff
@@ -95,3 +130,61 @@ export const getAllPluginsIndexedBySiteId = createSelector(
 	},
 	( state ) => [ getAllPluginsIndexedByPluginSlug( state ), getSiteIdsThatHavePlugins( state ) ]
 ) as { ( state: AppState ): { [ siteId: number ]: { [ pluginSlug: string ]: Plugin } } };
+
+export const getFilteredAndSortedPlugins = createSelector(
+	( state: AppState, siteIds: number[], pluginFilter?: PluginFilter ) => {
+		const allPluginsIndexedBySiteId = getAllPluginsIndexedBySiteId( state );
+
+		// Properties on the objects in allPluginsIndexedBySiteId will be modified and the
+		// selector memoization always returns the same object, so use cloneDeep to avoid
+		// altering it for everyone.
+		const allPluginsForSites: { [ pluginSlug: string ]: Plugin } = cloneDeep(
+			siteIds
+				.map( ( siteId: number ) => allPluginsIndexedBySiteId[ siteId ] )
+				.filter( Boolean )
+				.reduce( ( accumulator, current ) => ( { ...accumulator, ...current } ), {} )
+		);
+
+		// Filter the sites object on the plugins so that only data for the requested siteIds is present
+		for ( const pluginSlug of Object.keys( allPluginsForSites ) ) {
+			allPluginsForSites[ pluginSlug ].sites = Object.entries(
+				allPluginsForSites[ pluginSlug ].sites
+			)
+				.filter( ( [ siteId ] ) => siteIds.includes( Number( siteId ) ) )
+				.reduce( ( obj, [ siteId, site ] ) => {
+					obj[ siteId ] = site;
+					return obj;
+				}, {} as PluginSites );
+		}
+
+		// Filter the plugins using the pluginFilter if it is set
+		const pluginList =
+			pluginFilter && _filters[ pluginFilter ]
+				? Object.entries( allPluginsForSites )
+						.filter( ( [ , plugin ] ) => _filters[ pluginFilter ]( plugin ) )
+						.reduce( ( obj, [ pluginSlug, plugin ] ) => {
+							obj[ pluginSlug ] = plugin;
+							return obj;
+						}, {} as { [ pluginSlug: string ]: Plugin } )
+				: allPluginsForSites;
+
+		// Sort the plugins alphabetically by slug
+		const sortedPluginListEntries = Object.entries( pluginList ).sort(
+			( [ pluginSlugA ], [ pluginSlugB ] ) => {
+				const pluginSlugALower = pluginSlugA.toLowerCase();
+				const pluginSlugBLower = pluginSlugB.toLowerCase();
+				if ( pluginSlugALower < pluginSlugBLower ) {
+					return -1;
+				} else if ( pluginSlugALower > pluginSlugBLower ) {
+					return 1;
+				}
+				return 0;
+			}
+		);
+		return sortedPluginListEntries.map( ( [ , plugin ] ) => plugin );
```

Comment:
We could avoid this `map()` if we use `Object.values()` instead of `Object.entries()`, can't we?

Reply:
Yes, I made that change.